assets/js/base/context/cart-checkout/payment-methods/payment-method-data-context.js

```diff
@@ -241,12 +248,21 @@ export const PaymentMethodDataProvider = ( {
 				);
 			},
 			/**
-			 * @param {Object}           [paymentMethodData] Arbitrary payment method data to
-			 *                                               accompany the checkout.
-			 * @param {BillingData|null} [billingData]       The billing data accompanying the
-			 *                                               payment method.
+			 * @param {Object} [paymentMethodData] Arbitrary payment method data to
+			 * accompany the checkout.
+			 * @param {BillingData|null} [billingData] The billing data accompanying the
+			 * payment method.
+			 * @param {ShippingDataResponse|null} [shippingData] The shipping data accompanying the
+			 * payment method.
 			 */
-			success: ( paymentMethodData = {}, billingData = null ) => {
+			success: (
+				paymentMethodData = {},
+				billingData = null,
+				shippingData = null
+			) => {
+				if ( shippingData !== null && shippingData?.address ) {
+					setShippingAddress( shippingData.address );
+				}
```

Comment:
The payment success status dispatcher now will also set shipping address if present (because express payment methods could provide shipping address).